Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -319,36 +320,28 @@ public class PythonCompiler : SinglePassCompiler {
       return $"{(full ? dt.FullCompileName : dt.CompileName)}_{ctor.CompileName}";
     }
 
-    protected override IClassWriter DeclareNewtype(NewtypeDecl nt, ConcreteSyntaxTree wr) {
-      var cw = (ClassWriter)CreateClass(IdProtect(nt.EnclosingModuleDefinition.CompileName), IdName(nt), nt, wr);
+    protected IClassWriter DeclareType(TopLevelDecl d, SubsetTypeDecl.WKind witnessKind, Expression witness, ConcreteSyntaxTree wr) {
+      var cw = (ClassWriter)CreateClass(IdProtect(d.EnclosingModuleDefinition.CompileName), IdName(d), d, wr);
       var w = cw.MethodWriter;
-      var udt = UserDefinedType.FromTopLevelDecl(nt.tok, nt);
-      if (nt.WitnessKind == SubsetTypeDecl.WKind.Compiled) {
-        w.WriteLine("@staticmethod");
-        var block = w.NewBlockPy("def default():");
-        var wStmts = block.Fork();
-        block.Write("return ");
-        TrExpr(nt.Witness, block, false, wStmts);
+      var udt = UserDefinedType.FromTopLevelDecl(d.tok, d);
+      w.WriteLine("@staticmethod");
+      var block = w.NewBlockPy("def default():");
+      var wStmts = block.Fork();
+      block.Write("return ");
+      if (witnessKind == SubsetTypeDecl.WKind.Compiled) {
+        TrExpr(witness, block, false, wStmts);
       } else {
-        w.WriteLine($"default = staticmethod(lambda: {TypeInitializationValue(udt, wr, nt.tok, false, false)})");
+        block.Write(TypeInitializationValue(udt, wr, d.tok, false, false));
```

Comment:
This looks good, but this is another thing that we should do in Dafny in the long run; these two cases should be the same (we should auto-fill the witness as the TypeInitializationValue)